src/agent.rs

```diff
@@ -231,14 +288,17 @@ impl Discoverer for Context {
     }
 }
 
+impl Dispatchable for Context {}
+
 struct SlabResponder<AGN: Agent> {
-    slab: Shared<Slab<Callback<AGN::Output>>>,
+    slab: Shared<Slab<Option<Callback<AGN::Output>>>>,
 }
 
 impl<AGN: Agent> Responder<AGN> for SlabResponder<AGN> {
     fn response(&self, id: HandlerId, output: AGN::Output) {
-        let callback = self.slab.borrow().get(id.raw_id()).cloned();
-        if let Some(callback) = callback {
+        let callback: Option<Option<Callback<_>>> = self.slab.borrow().get(id.raw_id()).cloned();
+        if let Some(callback) = callback.and_then(std::convert::identity) {
+            // TODO replace with flatten when option_flattening (60258) stabilizes.
             callback.emit(output);
         } else {
             warn!("Id of handler not exists <slab>: {}", id.raw_id());
```

Comment:
I think we want a different warning message if an agent tries to send a response to a HandlerId with no callback

Reply:
Good, better warnings and avoids having to deal with `flatten` not existing yet.